### PR TITLE
fix(DB/Zone):improved Shadowmoon Valley

### PR DESCRIPTION
**Pull request by Knindzagxg**
## Changes Proposed:
- Missing SAI creature updates.
- Improvements to existing if possible.

## Tests Performed:
- Applied SQL without problems.
- Tested only Demon Hunters due quest chain and unique abilities.

## How to Test:
- Apply SQL(s).
- Restart Server (You can .reload creature_template for each creature as well, but will take time)
- Go to zone (Check PR name) and check creatures around - You can also spawn (.npc add) each by command.

## Source for Changes:
- [Wowhead](https://wowhead.com) Database and or comments.
- [WotlkDB](https://wotlkdb.com/) Multi checking trough Private DB's.
- [WowWiki](https://wowwiki-archive.fandom.com/wiki/Portal:Main) Can be useful for extra informations.
- [WowPedia](https://wow.gamepedia.com/Wowpedia) Cross checking trough comments if possible.
- [Retail](https://worldofwarcraft.com/en-gb/) In-game checking if previous don't have any required informations.

## Target Branch:
- [x] Master

## Operating System:
- Linux (Debian 10)

## How to Test AzerothCore PR's
When a PR is ready to be tested, it will be marked as **[Waiting to be tested]**.
You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:
http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by claudiodfc** [comment]
# I compiled a list with all the modified NPCs on this PR.
One thing I've found out is that many don't have any spells on aowow so every single one should be validated with other sources and with sniffs or youtube proof of their behaviour.
This PR and the other ones should not be merged without figuring out what issues does it fix if the SAI is correct and if no bugs are introduced.

- [ ] [Coilskar Muckwatcher](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19788)
- [ ] [Coilskar Cobra](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19784)
- [ ] [Coilskar Sorceress](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19767)
- [ ] [Coilskar Myrmidon](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19765)
- [ ] [Keeper of the Cistern](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=20795)
- [ ] [Ravenous Flayer Matriarch](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21205)
- [ ] [Coilskar Assassin](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21044)
- [ ] [Ravenous Flayer](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21196)
- [ ] [Shadow Serpent](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=23020)
- [ ] [Oronu the Elder](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21663)
- [ ] [Ashtongue Warrior <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21454)
- [ ] [Ashtongue Handler <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21803)
- [ ] [Elekk Demolisher](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21802)
- [ ] [Ashtongue Shaman <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21453)
- [ ] [Illidari Overseer <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21808)
- [ ] [Ashtongue Worker <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21455)
- [ ] [Smith Gorlunk](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22037)
- [ ] [Shadowhoof Summoner <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22859)
- [ ] [Illidari Dreadlord](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21166)
- [ ] [Spellbound Terrorguard](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21908)
- [ ] [Varedis <Illidari Demon Hunter>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21178)
- [ ] [Alandien <Illidari Demon Hunter>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21171)
- [ ] [Theras <Illidari Demon Hunter>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21168)
- [ ] [Netharel <Illidari Demon Hunter>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21164)
- [ ] [Sunfury Summoner](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21505)
- [ ] [Demon Hunter Supplicant](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21179)
- [ ] [Demon Hunter Initiate](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21180)
- [ ] [Sunfury Eradicator](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21742)
- [ ] [Zandras <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21827)
- [ ] [Dreadwarden <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19744)
- [ ] [Dreadwarden <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19801)
- [ ] [Illidari Jailor <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21520)
- [ ] [Rocknail Flayer](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21477)
- [ ] [Rocknail Ripper](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21478)
- [ ] [Netherskate](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21901)
- [ ] [Barash the Den Mother](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=23269)
- [ ] [Arvoar the Rapacious](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=23267)
- [ ] [Overmine Flayer](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=23264)
- [ ] [Dragonmaw Skybreaker](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22274)
- [ ] [Haalum <Corruptor of Air>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21711)
- [ ] [Illidari Watcher <The Crimson Sigil>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22093)
- [ ] [Corok the Mighty](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22011)
- [ ] [Val'zareq the Conqueror <The Crimson Sigil>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21979)
- [ ] [Eclipsion Blood Knight](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19795)
- [ ] [Son of Corok](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19824)
- [ ] [Illidari Slayer](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21639)
- [ ] [Skethyl Owl](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21450)
- [ ] [Enraged Earth Shard](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22115)
- [ ] [Infernal Soul](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19757)
- [ ] [Illidari Shadowstalker <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21337)
- [ ] [Lothros <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21928)
- [ ] [Illidari Painlasher <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19800)
- [ ] [Illidari Dreadbringer <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19799)
- [ ] [Illidari Satyr <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21656)
- [ ] [Illidari Shocktrooper <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=19802)
- [ ] [Felfire Diemetradon](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21408)

**Comment by pklloveyou** [comment]
First of all, it's better to have AI than none.

**Comment by Knindzagxg** [comment]
No private DB will provide you information of all spells.

Feel free to check every trough wowhead / login on retail and things like.

**Comment by claudiodfc** [comment]
> No private DB will provide you information of all spells.
> 
> Feel free to check every trough wowhead / login on retail and things like.

You say that, but you don't provide the source of your scripts. What's backing them up? Where did you get the behaviour from?

@pklloveyou There are NPCs that have no SmartAI intentionally. Don't have this mindset when contributing. X is never better than Y. Things have reasons for being as they are, even if they are bugged due to lack of information or motivation.

All I'm asking is the same everyone asks me whenever I send a PR, proof.

Let's not make exceptions for Staff members.

**Comment by Knindzagxg** [comment]
Like i said, when checking and making PR's you need to check multi sources.
And even that (Wowhead, Wiki, OpenWoW or any other) wont give you all information.

For example, you can check Demon Hunter SAI.
There is no information about spells they are using, and you have to dig even further (Example: Wowhead comments on specific Quests where that NPC will appear)

```Example:
Varedis: Flame Wave, fire dot on the ground
Theras: Spell Reflect
Alandien: Shadowfury, about 10 yard radius aoe knockback+damage
Netharel: (sorry cant remember, dont think anything significant though)```

Then cause it's not fully, you do more checking to discover they all have Metamorphosis and each have unique spell)

Beside that and 15 years of Retail experience i won't be able to provide.

Yet again, like i said above, all of checking need to be checked multi times, cause every single Private DB and WoWhead will have different results in spell ID's or etc.

Biggest issue that people blindly check Private DB's where during sniffs some old ID's were used and changed later on in patches 2.x or 3.x and or etc.

Some information about this can be found in Wowwiki if NPC is "important" but mostly that's not that case.
Last resort to get some extra information are Wowhead comments where you can check them by patch as well and see if there are changes.

**Comment by claudiodfc** [comment]
Good, now specify your changes like everyone does.

**Comment by Knindzagxg** [comment]
No, i'm not gonna type every change for every single NPC.

It's not single NPC upgrade but whole zone.

**Comment by FrancescoBorzi** [comment]
I think both of you have valid points.

On one hand, I understand that @Knindzagxg can't easily list all single change to every single NPC because it would take a while and he is working massively on them.

On the other hand, I agree with @claudiodfc about being careful to not introduce new bugs or at least non-critical ones.
However, being SAI quite self-contained (worse thing can happen: a single creature is bugged), I guess the best way to proceed is:

> Go to zone (Check PR name) and check creatures around - You can also spawn each by command.

do some quick checks in-game with those creatures and if everything seems right, we can merge it. If then a bug is found on a specific SAI row of a specific creature, we can always fix that later. But I would not wait that every single SAI row of every single creature is checked because it would take too long to do this manual job unless we have a volunteer of course.

For more delicate changes, for example, core changes or changes that in general could potentially negatively affect many things and break important features, we should be much more careful of course. But for self-contained DB-fixes, I would like to be flexible.

That said...

TLDR; can anyone test this zone in-game? Also @Knindzagxg please specify if you have tested it in-game already, when opening PRs like this ;)

**Comment by Knindzagxg** [comment]
About Skybreaker [Link](https://www.wowhead.com/npc=22274/dragonmaw-skybreaker#comments) -> Match from spells and comments (Patch 2.0.6)

**Comment by Knindzagxg** [comment]
[Ravenous Flayer](https://www.wowhead.com/npc=21196/ravenous-flayer#comments) - Confirmation in comments from 2.3.0 patch. 

[Ripper](https://www.wowhead.com/npc=21478/rocknail-ripper#abilities) - First change to spell in Cata when they changed attack power system - Indicates it was there during TBC/WOTLK.

[Smith Gorlunk](https://www.wowhead.com/spell=13444/sunder-armor#changelog) - Same as for Ripper. Changes have been made in Cata and they reduced debuff time from 30 to 20sec.

[Coilskar Assassin](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21044) - Removed SAI.
[Ashtongue Warrior <Servant of Illidan>](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21454) - Removed SAI.
[Enraged Earth Shard](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=22115) - Removed SAI.
[Rocknail Flayer](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=21477) - Removed SAI.

Will check / sort other during week.

PS. Edited all confirmations in your list as well.

**Comment by Knindzagxg** [comment]
- Ashtongue Handler - Will cast Enrage.
- Ashtongue Shaman - Will cast Fire Nova totem now.
- Arvoar the Rapacious - Will use Flay and Rock Shell. No enrage tho, cause trigger from Rock Shell is not scripted in core.
- Coilskar Muckwatcher - Updated spell ID.
- Netherskate - Will cast Nether Shock now.

- Terrifying Screech - Owl do cast it, it can miss. Check combat log.
Owl

- Lothros  - It does cast "Enrage" by spell name, but effect is Frenzy (Only haste) Wowhead confirms ID. Same for Illidari Dreadbringer as well.
- Ravenous Flayer Matriarch - https://www.wowhead.com/spell=36464/the-den-mothers-mark#comments - Sometimes we even have to read wowhead spell comments xD

- Spellbound Terrorguard - Wowhead confirm Hamsting for few creatures within TBC Exp.

- Eclipsion Blood Knight - Removed SAI.
- Shadow Serpent - Removed SAI.

--- 

Will need to check every Demon Hunter even more.
They all should have Metamorphosis -> Just need to confirm to they have same spells during that time.
On regular phase, each have 1 unique.

**Comment by Knindzagxg** [comment]
- Overmine Flayer - Updated.
- Illidari Slayer - Sorted SAI and run issue.
- Felfire Diemetradon - Was wrong ID that required creature to have mana -> Updated.

- Illidari Overseer  - It cast Mortal but it doesn't spam it. You prolly dodged attacked during tests xP
mortal

- Demon Hunter Initiate - https://www.wowhead.com/npc=21180/demon-hunter-initiate#comments Immolation confirmed in comments xP

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@Knindzagxg looks good to me

**Comment by Knindzagxg** [comment]
@FrancescoBorzi 

Still have to sort 4 Demon Hunters, but cant find all info that i need.
Should finish tomorrow.

**Comment by claudiodfc** [comment]
## Test results:
- [X] Ashtongue Handler casts Enrage;
- [X] Ashtongue Shaman casts Fire Nova totem;
- [X] Arvoar casts all his spells (Missing Enrage from unaura Rock Shell)
- [X] Coilskar Muckwatcher correct spell ID for Stab
- [X] Netherskate casts Nether shock
- [X] Overmine Flayer correct SAI (Missing Enrage from unaura Rock Shell)
- [X] Felfire Diemetradon correct SAI

## Notes:

- Illidari Slayers are still walking towards the player:
image

**Comment by claudiodfc** [comment]
I'm gonna open an issue for that Rock Shell so someone knows it exists and works on it

**Comment by claudiodfc** [comment]
https://github.com/azerothcore/azerothcore-wotlk/issues/5077 is open to fix Rock Shell

**Comment by Knindzagxg** [comment]
What you mean still walking? Removed run part on them (Disabled)

**Comment by Knindzagxg** [comment]
Updated all Demon Hunters and tested all in-game.

Acording to videos i found all have Metamorphosis -> https://www.youtube.com/watch?v=Er30umIXLnk&t=18s
Also @claudiodfc they dont cast any new spells during Metamorphosis, Book have wrong info for some reason.
And yeah, whats wrong with Ilidary Slayers? They should walk or run, cause i understood you to disable Run for them.

**Comment by Knindzagxg** [comment]
@claudiodfc check comment above and lemme know whats the problem
